components/rhs_thread/rhs_thread.tsx

```diff
@@ -114,6 +116,14 @@ export default class RhsThread extends React.Component<Props, State> {
             this.props.actions.getPostThread(this.props.selected.id);
         }
 
+        // if snapshot value, new posts have been added
+        // if at bottom of thread, continue to scroll after adding new post
+        if (curPostsArray.length > prevPostsArray.length) {
```

Comment:
Would this work if a reaction is added?

Reply:
Possibly not, I should inspect that array.

Reply:
It seems this check is redundant, because `snapshot` should not return true unless the posts array has mutated per the condition in `getSnapshotBeforeUpdate`

Reply:
You're correct, it doesn't account for reaction and it doesn't appear that `rhs_thread.tsx` has readily available access to the reaction data currently (would require polling Redux). I'm asking @deanwhillier about breaking this issue into another ticket. Since it could potentially increase the scope I think it makes sense to submit another incremental fix.

Reply:
With the buffer added, the thread will continue to scroll but it's not the best UX, obviously. http://g.recordit.co/lDdklYbGSJ.gif

Reply:
@sudheerDev I created a new issue for this bug: https://mattermost.atlassian.net/browse/MM-25550

Reply:
One more thing, This won't be just for cases like this but it could even be for post edit. 

If someone edits a post and adds a lot of content then it will push the RHS down but does not keep the user to the bottom